genesis/utils/usda.py

```diff
@@ -306,19 +191,165 @@ def parse_usd_material(material, surface, zipfiles):
+            gs.logger.warning(f"Replace symlink {asset_path} with real file {real_path}.")
+
+
+def decompress_usdz(usdz_path):
+    usdz_folder = mu.get_usd_zip_path(usdz_path)
+
+    # The first file in the package must be a native usd file.
+    # See https://openusd.org/docs/Usdz-File-Format-Specification.html
+    zip_files = Usd.ZipFile.Open(usdz_path)
+    zip_filelist = zip_files.GetFileNames()
+    root_file = zip_filelist[0]
+    if not root_file.lower().endswith(gs.options.morphs.USD_FORMAT[:-1]):
+        gs.raise_exception(f"Invalid usdz root file: {root_file}")
+    root_path = os.path.join(usdz_folder, root_file)
+
+    if not os.path.exists(root_path):
+        for file_name in zip_filelist:
+            file_data = io.BytesIO(zip_files.GetFile(file_name))
+            file_path = os.path.join(usdz_folder, file_name)
+            file_folder = os.path.dirname(file_path)
+            os.makedirs(file_folder, exist_ok=True)
+            with open(file_path, "wb") as out:
+                out.write(file_data.read())
+        gs.logger.warning(f"USDZ file {usdz_path} decompressed to {root_path}.")
+    else:
+        gs.logger.info(f"Decompressed assets detected and used: {root_path}.")
+    return root_path
+
+
+def parse_mesh_usd(path, group_by_material, scale, surface, bake_cache=True):
+    if path.lower().endswith(gs.options.morphs.USD_FORMAT[-1]):
+        path = decompress_usdz(path)
+
+    # detect bake file caches
+    is_bake_cache_found = False
+    baked_folder = mu.get_usd_bake_path(path)
+    baked_path = os.path.join(baked_folder, os.path.basename(path))
+    if bake_cache and os.path.exists(baked_path):
+        path = baked_path
+        is_bake_cache_found = True
+        gs.logger.info(f"Baked assets detected and used: {path}")
+
     stage = Usd.Stage.Open(path)
     scale *= UsdGeom.GetStageMetersPerUnit(stage)
     yup = UsdGeom.GetStageUpAxis(stage) == "Y"
     xform_cache = UsdGeom.XformCache()
 
     mesh_infos = mu.MeshInfoGroup()
-    materials = dict()
+    materials = {}
+    baked_materials = {}
 
+    # parse materials
+    for prim in stage.Traverse():
+        if prim.IsA(UsdShade.Material):
+            material_usd = UsdShade.Material(prim)
+            material_spec = prim.GetPrimStack()[-1]
+            material_id = material_spec.layer.identifier + material_spec.path.pathString
+            material_pack = materials.get(material_id, None)
+
+            if material_pack is None:
+                material, uv_name, require_bake = parse_usd_material(material_usd, surface)
+                materials[material_id] = (material, uv_name)
+                if not is_bake_cache_found and require_bake:
+                    baked_materials[material_id] = material_usd.GetPath()
+
+    if baked_materials:
+        baked_material_paths = list(map(str, baked_materials.values()))
+        baker_file = os.path.join(os.path.dirname(os.path.abspath(__file__)), "usda_bake.py")
```

Comment:
This is definitely not the right way to call python script. They should be implemented as proper python "entrypoints" to avoid requiring passing the full path of the file as you are doing.

Anyway, using subprocess should be avoided here, so you can complete forget about this.